frontend/src/metabase/admin/utils.js

```diff
@@ -17,3 +17,28 @@ export const createAdminRouteGuard = (routeKey, Component) => {
 
   return Wrapper(Component ?? (({ children }) => children));
 };
+
+const mapStateToProps = (state, props) => ({
+  adminItems: getAdminPaths(state),
+  path: props.location.pathname,
+});
+
+const mapDispatchToProps = {
+  push,
+  replace,
+};
+
+const _RedirectToAllowedSettings = ({ adminItems, replace }) => {
```

Comment:
moved from its own file in the containers/ directory to be closer to the router which is the only place it is ever used